packages/itensor/quri_parts/itensor/estimator.py

```diff
@@ -0,0 +1,205 @@
+from typing import TYPE_CHECKING, Any, Callable, NamedTuple, Optional
+
+import juliacall
+import numpy as np
+from juliacall import Main as jl
+from typing_extensions import TypeAlias
+
+from quri_parts.core.estimator import (
+    ConcurrentParametricQuantumEstimator,
+    ConcurrentQuantumEstimator,
+    Estimatable,
+    Estimate,
+    ParametricQuantumEstimator,
+    QuantumEstimator,
+    create_parametric_estimator,
+)
+from quri_parts.core.operator import zero
+from quri_parts.core.state import CircuitQuantumState, ParametricCircuitQuantumState
+from quri_parts.core.utils.concurrent import execute_concurrently
+
+from .circuit import convert_circuit
+from .operator import convert_operator
+
+if TYPE_CHECKING:
+    from concurrent.futures import Executor
+
+abs_dir = os.path.dirname(os.path.abspath(__file__))
+library_path = os.path.join(abs_dir, "library.jl")
+include_statement = 'include("' + library_path + '")'
+jl.seval(include_statement)
+
+
+class _Estimate(NamedTuple):
+    value: complex
+    error: float = np.nan
+
+
+#: A type alias for state classes supported by ITensor estimators.
+#: ITensor estimators support circuit states.
+ITensorStateT: TypeAlias = CircuitQuantumState
+
+#: A type alias for parametric state classes supported by ITensor estimators.
+#: ITensor estimators support circuit states.
+ITensorParametricStateT: TypeAlias = ParametricCircuitQuantumState
+
+
+def _estimate(operator: Estimatable, state: ITensorStateT) -> Estimate[complex]:
+    if operator == zero():
+        return _Estimate(value=0.0)
+    qubits = state.qubit_count
+    s: juliacall.VectorValue = jl.siteinds("Qubit", qubits)
+    psi: juliacall.AnyValue = jl.initState(s, qubits)
+
+    # create ITensor circuit
+    circuit = convert_circuit(state.circuit, s)
+
+    # create ITensor operator
+    op = convert_operator(operator, s)
+
+    # calculate expectation value
+    psi = jl.apply(circuit, psi)
+    exp: float = jl.expectation(psi, op)
+
+    return _Estimate(value=exp)
+
+
+def create_itensor_mps_estimator() -> QuantumEstimator[ITensorStateT]:
+    """Returns a :class:`~QuantumEstimator` that uses ITensor MPS simulator to
+    calculate expectation values."""
+
+    return _estimate
+
+
+def _sequential_estimate(
+    _: Any, op_state_tuples: Sequence[tuple[Estimatable, ITensorStateT]]
+) -> Sequence[Estimate[complex]]:
+    return [_estimate(operator, state) for operator, state in op_state_tuples]
+
+
+def _sequential_estimate_single_state(
+    state: ITensorStateT, operators: Sequence[Estimatable]
+) -> Sequence[Estimate[complex]]:
+    qubits = state.qubit_count
+    s: juliacall.VectorValue = jl.siteinds("Qubit", qubits)
+    psi: juliacall.AnyValue = jl.initState(s, qubits)
+    circuit = convert_circuit(state.circuit, s)
+    psi = jl.apply(circuit, psi)
+    results = []
+    for op in operators:
+        itensor_op = convert_operator(op, s)
+        results.append(_Estimate(value=jl.expectation(psi, itensor_op)))
```

Comment:
This part causes error when there is an empty `Operator` in `operators`.



```python
concurrent_mps_operator([Operator({...}), {}, …], [state])
```

   → JuliaError: OpSum has no terms
```suggestion
    for op in operators:
        if op == zero():
            results.append(_Estimate(value=0.0))
            continue
        itensor_op = convert_operator(op, s)
        results.append(_Estimate(value=jl.expectation(psi, itensor_op)))
```

Reply:
Oh! Thank you!